rpcs/blockService.go

```diff
@@ -326,3 +396,16 @@ func RawBlockBytes(l *data.Ledger, round basics.Round) ([]byte, error) {
 		Certificate: cert,
 	}), nil
 }
+
+// FormatBlockQuery formats a block request query for the given network and round number
+func FormatBlockQuery(round uint64, parsedURL string, net network.GossipNode) string {
+	return net.SubstituteGenesisID(path.Join(parsedURL, "/v1/{genesisID}/block/"+strconv.FormatUint(uint64(round), 36)))
+}
+
+func makeFallbackEndpoints(customFallbackEndpoints string) (fe fallbackEndpoints) {
+	if customFallbackEndpoints == "" {
+		return
+	}
+	fe = fallbackEndpoints{endpoints: strings.Split(customFallbackEndpoints, ",")}
```

Comment:
use `network.ParseHostOrURL` to verify the endpoint correctness. ( you can extract the `Host` component from the URL object) .
If any of the entries in the list is invalid, write a warning to the log.